apps/pipeline/lib/pipeline/writer/table_writer/compaction.ex

```diff
@@ -66,4 +67,17 @@ defmodule Pipeline.Writer.TableWriter.Compaction do
       results.rows
     end
   end
+
+  def add_dataset_record_event_count(table_name, old_row_count, new_row_count) do
+    if !String.ends_with?(table_name, "__json") do
+      [
+        table_name: table_name,
+        old_row_count: old_row_count
```

Comment:
This should be `row_count` right? We don't care about old vs new so long as they are the same. To that end, what happens when the compaction is rejected for row counts not matching? Won't this record the wrong one?